src/wristpy/core/models.py

```diff
@@ -37,3 +53,23 @@ class WatchData(BaseModel):
     battery: Optional[Measurement] = None
     capsense: Optional[Measurement] = None
     temperature: Optional[Measurement] = None
+
+    @field_validator("acceleration")
+    def validate_acceleration(cls, v: Measurement) -> Measurement:
+        """Validate the acceleration data.
+
+        Ensure that the acceleration data is a 2D array with 3 columns.
+
+        Args:
+            cls: The class.
+            v: The acceleration data to validate.
+
+        Returns:
+            v: The acceleration data if it is valid.
+
+        Raises:
+            ValueError: If the acceleration data is not a 2D array with 3 columns.
+        """
+        if len(v.measurements.shape) < 2 or v.measurements.shape[1] != 3:
```

Comment:
This should be `len(v.measurements.shape) == 2`.

Reply:
actually this is v.measurements.ndim !=2, I'm pretty sure